### PR TITLE
Fix Series/Dataframe Mixed Arithmetic

### PR DESCRIPTION
**Pull request by brandon-b-miller**
Fixes https://github.com/rapidsai/cudf/issues/7385

### CODE REVIEW

File: python/cudf/cudf/tests/test_dataframe.py [hunk 1]
```diff
@@ -5008,30 +5008,33 @@ def test_cov_nans():
...
         operator.mod,
         operator.pow,
-        operator.eq,
-        operator.lt,
-        operator.le,
-        operator.gt,
-        operator.ge,
-        operator.ne,
+        # TODO: Remove XFAILs after PR is merged
+        pytest.param(operator.eq, marks=pytest.mark.xfail()),
```
Comment (galipremsagar) [inline python/cudf/cudf/tests/test_dataframe.py:5012]: Can we remove these xfails? If not should we put in the url to PR here?

Comment (brandon-b-miller) [reply python/cudf/cudf/tests/test_dataframe.py]: this was dependent on PR https://github.com/rapidsai/cudf/pull/7490, and was removed 👍

**Review by galipremsagar — APPROVED** [review]

**Review by isVoid — APPROVED** [review]

**Comment by brandon-b-miller** [comment]
rerun tests

**Comment by brandon-b-miller** [comment]
@gpucibot merge

**Pull request merged**